pkg/segment/query/processor/evalcommand_test.go

```diff
@@ -0,0 +1,164 @@
+					ConcatExpr: &structs.ConcatExpr{
+						Atoms: []*structs.ConcatAtom{
+							{IsField: false, Value: "Massa"},
+							{IsField: false, Value: "chusetts"},
+						},
+					},
+				},
+			},
+			&structs.ValueExpr{
+				ValueExprMode: structs.VEMStringExpr,
+				StringExpr: &structs.StringExpr{
+					StringExprMode: structs.SEMRawString,
+					RawString:      "RAW_STRING",
+				},
+			},
+			&structs.ValueExpr{
+				ValueExprMode: structs.VEMNumericExpr,
+				NumericExpr: &structs.NumericExpr{
+					NumericExprMode: structs.NEMNumericExpr,
+					Op:              "+",
+					Left: &structs.NumericExpr{
+						NumericExprMode: structs.NEMNumber,
+						IsTerminal:      true,
+						ValueIsField:    false,
+						Value:           "99",
+					},
+					Right: &structs.NumericExpr{
+						NumericExprMode: structs.NEMNumber,
+						IsTerminal:      true,
+						ValueIsField:    false,
+						Value:           "1",
+					},
+				},
+			},
+		},
+	}
+
+	evalProcessor := &evalProcessor{
+		options: &structs.EvalExpr{
+			ValueExpr: &structs.ValueExpr{
+				ValueExprMode: structs.VEMConditionExpr,
+				ConditionExpr: &structs.ConditionExpr{
+					Op:       "if",
+					BoolExpr: boolExpr,
+					TrueValue: &structs.ValueExpr{
+						ValueExprMode: structs.VEMNumericExpr,
+						NumericExpr: &structs.NumericExpr{
+							NumericExprMode: structs.NEMNumberField,
+							IsTerminal:      true,
+							ValueIsField:    true,
+							Value:           "numField",
+						},
+					},
+					FalseValue: &structs.ValueExpr{
+						ValueExprMode: structs.VEMStringExpr,
+						StringExpr: &structs.StringExpr{
+							StringExprMode: structs.SEMField,
+							FieldName:      "country",
+						},
+					},
+				},
+			},
+			FieldName: "evalField",
+		},
+	}
+
+	inputIqr := iqr.NewIQR(0)
+	knownValues := map[string][]utils.CValueEnclosure{
+		"state": {
+			utils.CValueEnclosure{Dtype: utils.SS_DT_STRING, CVal: "Massachusetts"},
+			utils.CValueEnclosure{Dtype: utils.SS_DT_STRING, CVal: "California"},
+			utils.CValueEnclosure{Dtype: utils.SS_DT_STRING, CVal: "New York"},
+			utils.CValueEnclosure{Dtype: utils.SS_DT_STRING, CVal: "RAW_STRING"},
+		},
+		"numField": {
+			utils.CValueEnclosure{Dtype: utils.SS_DT_SIGNED_NUM, CVal: int64(1)},
+			utils.CValueEnclosure{Dtype: utils.SS_DT_SIGNED_NUM, CVal: int64(2)},
+			utils.CValueEnclosure{Dtype: utils.SS_DT_SIGNED_NUM, CVal: int64(3)},
+			utils.CValueEnclosure{Dtype: utils.SS_DT_SIGNED_NUM, CVal: int64(4)},
+		},
+		"country": {
+			utils.CValueEnclosure{Dtype: utils.SS_DT_STRING, CVal: "USA"},
+			utils.CValueEnclosure{Dtype: utils.SS_DT_STRING, CVal: "India"},
+			utils.CValueEnclosure{Dtype: utils.SS_DT_STRING, CVal: "China"},
+			utils.CValueEnclosure{Dtype: utils.SS_DT_STRING, CVal: "Japan"},
+		},
+	}
+
+	evalField := []utils.CValueEnclosure{
+		{Dtype: utils.SS_DT_STRING, CVal: "1"},
+		{Dtype: utils.SS_DT_STRING, CVal: "India"},
+		{Dtype: utils.SS_DT_STRING, CVal: "China"},
```

Comment:
I think these are your expected values, right? Renaming to `expected` or `expectedEvalField` or similar would be helpful

Reply:
addressed in the latest commit.